frontend/packages/console-shared/src/types/pod.ts

```diff
@@ -1,6 +1,6 @@
 import { DeploymentKind, K8sResourceKind, PodKind, PodPhase } from '@console/internal/module/k8s';
 import { FirehoseResult } from '@console/internal/components/utils';
-import { AllPodStatus } from '../constants';
+import { AllPodStatus } from '../constants/pod';
```

Comment:
```suggestion
import { AllPodStatus } from '../constants';
```
Any reason we don't want to take directly from constants?

Reply:
I am just trying to establish a rule to consistently follow across the codebase.
When importing code inside the package, I lean towards not using the re-exports to limit potential circular dependencies. But I am happy to change that.

Reply:
I don't have a strong opinion here, but if we choose one way or another, let's follow that consistently.

Assuming the current preference for imports outside the shared pkg (prefer `@console/shared`), the preference for imports inside the shared pkg would better match what @andrewballantyne suggests.

But @jtomasek is right that adding more layers of indirection, i.e. re-(re-(re..))-exports add more complexity to the overall module dependency graph and therefore risk potential for cyclic dependencies.

We should first converge to one consistent way and then improve from there. I really like @christianvogt's proposal on having `lib` as import-able API which organizes the whole pkg into logical sub-systems.